libs/langchain/langchain/document_loaders/parsers/pdf_docai.py

```diff
@@ -0,0 +1,217 @@
+"""Module contains a PDF parser based on DocAI from Google Cloud.
+
+You need to install two libraries to use this parser:
+pip install google-cloud-documentai
+pip install google-cloud-documentai-toolbox
+"""
+import time
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Iterator, List, Optional
+
+from google.api_core.operation import Operation
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.document_loaders.blob_loaders import Blob
+
+if TYPE_CHECKING:
+    from google.cloud.documentai import DocumentProcessorServiceClient
+
+
+@dataclass
+class DocAIParsingResults:
+    """A dataclass to store DocAI parsing results."""
+
+    source_path: str
+    parsed_path: str
+
+
+class DocAIPdfLoader(BaseBlobParser):
+    def __init__(
+        self,
```

Comment:
MAJOR: Could you allow a user to pass a pre-instantiated client here in addition to allowing a user to only pass some config paramets? 

The life time of the blob parser may be different from expected and we want the user to be able to control the instantiation in those cases.

Reply:
Done

Reply:
I meant something along these lines

```python

def __init__(self, 
 *, 
 client: Optional[]  = None,
  location: Optional[str] = None,
  processor_name: Optinonal[str] = None,
) -> None:
  if client and location and process_name: raise ValueError('must specify either client or the other things but not both')
  if not client and not (location and process_name): raise ValueError('must either client or the other things')

  if client: 
     self._client = client 
  else:
    self._client = instantiate client using parameters # <-- Convenience for users
```

Here, after initialization is complete, there will be no code path under which self._client is still a `None`.

This will allow a user to pass a long a client with a different life-cycle than the parser if need be, but it'll also allow users that don't care to use the convenient instantiation method provided by init. 

Passing the client is more important than the config parameters to allow life cycle control

Reply:
Done